src/Prometheus.Client.HttpClient/HttpClientMetricsExtensions.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.DependencyInjection;
+using Prometheus.Client.HttpClient.MessageHandlers;
+using Prometheus.Client.HttpClient.Options;
+
+namespace Prometheus.Client.HttpClient
+{
+    public static class HttpClientMetricsExtensions
+    {
+        /// <summary>
+        /// Configures the HttpClient pipeline to collect Prometheus metrics.
+        /// </summary>
+        public static IHttpClientBuilder UseHttpClientMetrics(this IHttpClientBuilder builder, Action<HttpClientMetricsOptions> configure)
+        {
+            var options = new HttpClientMetricsOptions();
+
+            configure?.Invoke(options);
+
+            builder.UseHttpClientMetrics(options);
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Configures the HttpClient pipeline to collect Prometheus metrics.
+        /// </summary>
+        public static IHttpClientBuilder UseHttpClientMetrics(this IHttpClientBuilder builder, HttpClientMetricsOptions options = null)
+        {
+            options ??= new HttpClientMetricsOptions();
+
+            var metricFactory = new MetricFactory(options.CollectorRegistry ?? new Collectors.CollectorRegistry());
```

Comment:
Something like this:
```c#
options.CollectorRegistry
                ??= (ICollectorRegistry)serviceProvider.GetService(typeof(ICollectorRegistry))
                    ?? Metrics.DefaultCollectorRegistry;
```

I guess you can access to `ServiceProvider` only inside `AddHttpMessageHandler`.

Reply:
Example: https://github.com/prom-client-net/prom-client-httprequestdurations/blob/8032a42be1626f77266c4f458075f6aa805bda53/src/HttpRequestDurationsMiddlewareBuilderExtensions.cs#L28C13-L30C57

Reply:
Can create `MetricFactory` for each handler, but should be the same `CollectorRegisry`.

Reply:
Done but kept same `metricFactory` for now.

Reply:
@manan-habib Get ServiceProvider from `IHttpClientBuilder`.

Reply:
@phnx47 
Please ignore this change for now, gonna be a new change committing shortly.

Reply:
@phnx47 
Updated now.